vm.c

```diff
@@ -2648,20 +2648,25 @@ th_init(rb_thread_t *th, VALUE self)
     th->self = self;
     rb_threadptr_root_fiber_setup(th);
 
-    {
-	/* vm_stack_size is word number.
-	 * th->vm->default_params.thread_vm_stack_size is byte size. */
-	size_t size = th->vm->default_params.thread_vm_stack_size / sizeof(VALUE);
-	rb_ec_set_vm_stack(th->ec, rb_thread_recycle_stack(size), size);
+    if (self == 0) {
```

Comment:
what mean?: `self == 0`

Reply:
It's how `th_init` is called for main thread. I wondered if we should change to `Qnil`, but I think it's outside scope for this PR.